api/document/js/fetch-doc.ts

```diff
@@ -8,10 +9,25 @@ export function convertNode(node) {
   return NODE_TYPE_CONVERTERS[nodeType](node);
 }
 
+export function convertContent(content, marks: Mark[]): Node[] {
+  if (content.content_type === '__text__') {
+    const text = (content.text || '').replace(/\s+/g, ' ');
+    return [schema.text(text, marks)];
+  }
+  const contentType = content.content_type in CONTENT_TYPE_CONVERTERS ?
+    content.content_type : 'unimplemented_mark';
+  const mark = CONTENT_TYPE_CONVERTERS[contentType](content);
+  const updatedMarks = marks.concat([mark]);
+  const children = (content.inlines || []).map(child =>
+    convertContent(child, updatedMarks));
+  return [].concat(...children);
```

Comment:
I wonder if adding a type annotation to `children` here, purely for the sake of documentation, would be useful?  Or maybe rename it to `childNodeLists`?

Also, I find `Array.prototype.concat`'s behavior really confusing because of how it does a different thing depending on whether its arguments are arrays or not.  I wonder if it'd help to assign this value to a well-named variable, like say, `flattenedChildren`, or maybe factor it out into a well-named utility function like `shallowFlatten` or something... eh, whatever, I'm probably just dense.

Reply:
Can do, good thoughts, thanks @toolness

Reply:
I'll throw in lodash; it's well known and likely to come in handy later.